api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto

```diff
@@ -168,6 +168,11 @@ message AuthorizationRequest {
   // 2. *Content-Length* will be set to 0 and the request to the authorization service will not have
   // a message body.
   //
+  // .. attention::
+  //
+  //   Since header keys are lowercase strings, please make sure the supplied :ref:`patterns
+  //   <envoy_api_field_type.matcher.ListStringMatcher.patterns>` are intended to match lowercase
+  //   strings.
```

Comment:
I think we have to deprecate this field or at least deprecate the behavior? This could break someone and/or cause a security issue I think? @htuch WDYT is the best way of handling this?

Reply:
I think we have the situation where the API did not promise a specific behavior, so we can safely change it without implying a breaking change. Realistically, some folks may depend on this, so I think we should do a runtime feature flag and sunset the lower case matcher that way.

Reply:
+1 SGTM to do a runtime flag

/wait